tests/PHPStan/Analyser/data/count-maybe.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace CountMaybe;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param float|int[] $maybeCountable
+ */
+function doFoo1($maybeCountable, int $mode): void
+{
+	if (count($maybeCountable, $mode) > 0) {
+		assertType('non-empty-array<int>', $maybeCountable);
+	} else {
+		assertType('array<int>|float', $maybeCountable);
+	}
+}
+
+function doBar1(float $notCountable, int $mode): void
+{
+	if (count($notCountable, $mode) > 0) {
+		assertType('float', $notCountable);
+	} else {
+		assertType('float', $notCountable);
+	}
+}
+
+/**
+ * @param array|int $maybeMode
+ */
+function doBar2(float $notCountable, $maybeMode): void
+{
+	if (count($notCountable, $maybeMode) > 0) {
+		assertType('float', $notCountable);
+	} else {
+		assertType('float', $notCountable);
+	}
+}
+
+function doBar3(float $notCountable, float $invalidMode): void
+{
+	if (count($notCountable, $invalidMode) > 0) {
+		assertType('float', $notCountable);
```

Comment:
should we resolve to a `*ERROR*` on invalid mode? I was not sure, as we already have a error at the count level because of the invalid argument type...?